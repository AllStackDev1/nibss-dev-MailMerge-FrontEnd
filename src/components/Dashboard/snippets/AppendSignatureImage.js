import React from 'react';
import Trigger from "./append-signature/Trigger";
import styled from 'styled-components';
import { getImageSize } from 'helpers';
import decode from 'jwt-decode';

const AppendSignatureImage = ({ imageRef, isNumPagesSet, width75percent, document, setImageError, setDocument,
    user, userToken, signDocumentConst, setModal }) => {
    const calculateOffset = async e => {
        const imageSize = await getImageSize(e.target);

        document.document.signatories.forEach((signatory, i) => {
            const documentCopy = Object.assign({}, document);
            documentCopy.document.signatories[i] = {
                ...signatory,
                absolute_x_coordinate: (signatory.x_coordinate / imageSize.width) * imageRef.current.offsetWidth,
                absolute_y_coordinate: (signatory.y_coordinate / imageSize.height) * imageRef.current.offsetHeight
            }
            setDocument(documentCopy);
        })
    }

    return (
        <PageContainer className={`${isNumPagesSet ? width75percent : ''}`}>
            <img ref={imageRef} onLoad={calculateOffset} onError={() => setImageError(true)} src={document?.document.file}
                className="full-width right-margin-10" alt="NIBSS Upload Document" />
            {document?.document.signatories
                .filter(signatory => signatory.email === (user?.data?.email || decode(userToken)?.data?.email))
                .map((signatory, index) =>
                    signatory.absolute_x_coordinate !== undefined ?
                        <Trigger signatory={signatory} signDocumentConst={signDocumentConst} index={index} setModal={setModal}
                            user={user} userToken={userToken} />
                        : <div></div>
                )}
        </PageContainer>
    )
}

const PageContainer = styled.div`
                border: 1px solid #CCC !important;
                &:hover {
                    border: 1px dashed #d8d8d8 !important;
                }
            `;

export default AppendSignatureImage;
