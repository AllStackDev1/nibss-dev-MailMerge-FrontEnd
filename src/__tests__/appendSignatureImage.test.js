import React from 'react';
import { shallow } from 'enzyme';
import AppendSignatureImage from "../components/Dashboard/snippets/AppendSignatureImage"



const shallowSetup = (props = {}) => {

    return shallow(<AppendSignatureImage {...props} />)
}

it('should render with error', () => {
    const wrapper = shallowSetup();
    expect(wrapper).toBeTruthy();
})

it("shouls render snapshot", () => {

    const wrapper = shallowSetup();
    expect(wrapper).toMatchSnapshot();

})

it("should load image error", () => {

    const wrapper = shallowSetup({
        setImageError: jest.fn()
    });
    const signatureImg = wrapper.find(".right-margin-10");

    signatureImg.prop("onError")();
    expect(signatureImg.prop("onError")).toBeTruthy();

    signatureImg.prop("onLoad")({ target: "200" })

})

