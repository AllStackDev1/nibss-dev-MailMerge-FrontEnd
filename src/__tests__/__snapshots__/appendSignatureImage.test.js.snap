// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shouls render snapshot 1`] = `ShallowWrapper {}`;
